api/renewer_test.go

```diff
@@ -233,3 +237,54 @@ func TestLifetimeWatcher(t *testing.T) {
 		})
 	}
 }
+
+// TestCalcSleepPeriod uses property based testing to evaluate the calculateSleepDuration
+// function of LifeTimeWatchers, but also incidentally tests "calculateGrace".
+// This is on account of "calculateSleepDuration" performing the "calculateGrace"
+// function in particular instances.
+// Both of these functions support the vital functionality of the LifeTimeWatcher
+// and therefore should be tested rigorously.
+func TestCalcSleepPeriod(t *testing.T) {
+	c := quick.Config{
+		MaxCount: 1000,
+		Values: func(values []reflect.Value, r *rand.Rand) {
+			// total lease duration
+			leaseDuration := r.Intn(math.MaxInt64)
+			priorDuration := r.Intn(leaseDuration)
+			remainingLeaseDuration := r.Intn(priorDuration)
+			increment := r.Intn(remainingLeaseDuration)
+
+			values[0] = reflect.ValueOf(r)
+			values[1] = reflect.ValueOf(time.Duration(leaseDuration))
+
+			// prior lease duration
+			values[2] = reflect.ValueOf(time.Duration(priorDuration))
+			// remaining lease duration
+			values[3] = reflect.ValueOf(time.Duration(remainingLeaseDuration))
+
+			// increment
+			// integer truncation... could be interesting.
+			values[4] = reflect.ValueOf(increment)
+		},
+	}
+
+	// tests that "calculateSleepDuration" will always return a value less than
+	// the remaining lease duration given a random leaseDuration, priorDuration, remainingLeaseDuration, and increment.
+	// Inputs are generated so that:
+	// leaseDuration > priorDuration > remainingLeaseDuration
+	// and remainingLeaseDuration > increment
+	if err := quick.Check(func(r *rand.Rand, leaseDuration, priorDuration, remainingLeaseDuration time.Duration, increment int) bool {
+		lw := LifetimeWatcher{
+			grace:     0,
+			increment: increment,
+			random:    r,
+		}
+
+		lw.calculateGrace(leaseDuration, time.Duration(increment))
```

Comment:
I believe we should be passing in the `remainingLeaseDuration` here: 

```suggestion
		lw.calculateGrace(remainingLeaseDuration, time.Duration(increment))
```